airflow/providers/amazon/aws/triggers/base_trigger.py

```diff
@@ -0,0 +1,129 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Any, AsyncIterator
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsGenericHook
+from airflow.providers.amazon.aws.utils.waiter_with_logging import async_wait
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class AwsBaseWaiterTrigger(BaseTrigger):
+    """
+    Base class for all AWS Triggers that follow the "standard" model of just waiting on a waiter.
+
+    Subclasses need to implement the hook() method.
+
+    :param serialized_fields: Fields that are specific to the subclass trigger and need to be serialized
+        to be passed to the __init__ method on deserialization.
+        format: {<parameter_name>: <parameter_value>}
```

Comment:
This should specify which fields are serialized by default. For example, we don't want someone passing `aws_conn_id` as a serialized field, because even though it needs to be serialized, it is done by default. What do you think?

Reply:
hmm, yeah. From the way it's used, it'd be mostly harmless, just unnecessary. I can add it to the comments.